src/clients/SnapshotApi.ts

```diff
@@ -0,0 +1,169 @@
+    }
+    return new Wallet(SNAPSHOT_PRIVATE_KEY)
+  }
+
+  private static getSpace() {
+    if (!SNAPSHOT_SPACE) {
+      throw new Error('Failed to determine snapshot space. Please check GATSBY_SNAPSHOT_SPACE env is defined')
+    }
+    return SNAPSHOT_SPACE
+  }
+
+  private static getSnapshotAddress() {
+    if (!SNAPSHOT_ADDRESS) {
+      throw new Error('Failed to determine snapshot address. Please check GATSBY_SNAPSHOT_ADDRESS env is defined')
+    }
+    return SNAPSHOT_ADDRESS
+  }
+
+  static get() {
+    return this.from(env('SNAPSHOT_API', this.Url))
+  }
+
+  async createProposal(
+    proposal: ProposalInCreation,
+    proposalTitle: string,
+    proposalBody: string,
+    proposalLifespan: ProposalLifespan,
+    blockNumber: number
+  ): Promise<SnapshotReceipt> {
+    const proposalMessage = await this.createProposalMessage(
+      proposal,
+      proposalTitle,
+      proposalBody,
+      proposalLifespan,
+      blockNumber
+    )
+    return (await this.client.proposal(this.account, this.address, proposalMessage)) as SnapshotReceipt
+  }
+
+  private async createProposalMessage(
+    proposal: ProposalInCreation,
+    proposalTitle: string,
+    proposalBody: string,
+    proposalLifespan: ProposalLifespan,
+    blockNumber: number
+  ) {
+    let snapshotProposal: Proposal
+    try {
+      snapshotProposal = {
+        space: this.spaceName,
+        type: SNAPSHOT_PROPOSAL_TYPE,
+        title: proposalTitle,
+        body: proposalBody,
+        choices: proposal.configuration.choices as string[],
+        start: SnapshotApi.toSnapshotTimestamp(proposalLifespan.start.getTime()),
+        end: SnapshotApi.toSnapshotTimestamp(proposalLifespan.end.getTime()),
+        snapshot: blockNumber,
+        plugins: JSON.stringify({}),
+        app: SNAPSHOT_APP_NAME,
+        discussion: '',
+      }
+    } catch (err) {
+      throw new Error('Error building the proposal creation message', err as Error)
+    }
+
+    return snapshotProposal
+  }
+
+  public async removeProposal(snapshotId: string) {
+    const cancelProposalMessage: CancelProposal = {
+      space: this.spaceName,
+      timestamp: SnapshotApi.toSnapshotTimestamp(Time.from().getTime()),
+      proposal: snapshotId,
+    }
+    return (await this.client.cancelProposal(this.account, this.address, cancelProposalMessage)) as SnapshotReceipt
+  }
+
+  async castVote(
+    account: Web3Provider | Wallet,
+    address: string,
+    proposalSnapshotId: string,
+    choiceNumber: number
+  ): Promise<SnapshotReceipt> {
+    const voteMessage = {
+      space: this.spaceName,
+      proposal: proposalSnapshotId,
+      type: SNAPSHOT_PROPOSAL_TYPE,
+      choice: choiceNumber,
+      app: SNAPSHOT_APP_NAME,
+    }
+    return (await this.client.vote(account, address, voteMessage)) as SnapshotReceipt
+  }
+
+  async getScores(addresses: string[], blockNumber?: number | string) {
+    addresses = addresses.map((addr) => addr.toLowerCase())
```

Comment:
I'd add the map to a new variable to avoid replacing the initial variable.
`const formattedAddresses = addresses.map((addr) => addr.toLowerCase())`